packages/gh/src/commands/deploy-preview.ts

```
@@ -0,0 +1,112 @@
+import { Command, flags } from '@oclif/command';
+import { request } from '@octokit/request';
+import execa from 'execa';
+
+function getPreviewURL(text: string): string {
+  const match = /Website Draft URL: (.+)/.exec(text);
+
+  if (!match) {
+    throw new Error('Cannot find preview url from logs');
+  }
+
+  return match[1];
+}
+
+export default class DeployPreview extends Command {
+  static description = 'deploy preview';
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+    dir: flags.string({ char: 'd', description: 'build dir', required: true }),
+    alias: flags.string({
+      char: 'a',
+      description: 'deploy alias',
+      required: true,
+    }),
+  };
+
+  static args = [];
+
+  async run() {
+    const {
+      flags: { dir, alias },
+    } = this.parse(DeployPreview);
+
+    const {
+      GITHUB_SHA,
+      GITHUB_TOKEN,
+      GITHUB_REPOSITORY,
+      GITHUB_PULL_REQUEST_NUMBER,
+    } = process.env;
+
+    if (!GITHUB_SHA) {
+      throw new Error('Please provide `GITHUB_SHA`.');
+    }
+
+    if (!GITHUB_TOKEN) {
+      throw new Error('Please provide `GITHUB_TOKEN`.');
+    }
+
+    if (!GITHUB_REPOSITORY) {
+      throw new Error('Please provide `GITHUB_REPOSITORY`.');
+    }
+
+    if (!GITHUB_PULL_REQUEST_NUMBER) {
+      throw new Error('Please provide `GITHUB_PULL_REQUEST_NUMBER`.');
+    }
+
+    const [owner, repo] = GITHUB_REPOSITORY.split('/');
+    const issueNumber = parseInt(GITHUB_PULL_REQUEST_NUMBER);
+
+    const { data: comments } = await request(
+      'GET /repos/:owner/:repo/issues/:issue_number/comments',
+      {
+        repo,
+        owner,
+        issue_number: issueNumber,
+        headers: { authorization: `Token ${GITHUB_TOKEN}` },
+      },
+    );
+
+    const { stdout } = await execa('yarn', [
```

this will not output anything from the `netlify deploy` script, so it will silently fail. 

You still want to pipe the output to the current `stdout` and `stderr`:
```ts
const deployProcess = execa('yarn', […]);

deployProcess.stdout.pipe(process.stdout);
deployProcess.stderr.pipe(process.stderr);

const { stdout } = await deployProcess;
```